### ISSUE TITLE
Conditional boolean property access should evaluate to false in a condition when left-hand side is null

### ISSUE THREAD

**Issue by jmagman** [dart-lang/language#1086]
```dart
class Foo {
  bool isBar;
}

void checkBar() {
  var foo = Foo();
  if (foo?.isBar) { // null_aware_in_condition warning happens here
    print("I'm a bar");
  }
}
```
Analyzer complains with `null_aware_in_condition`:
```
The value of the '?.' operator can be 'null', which isn't appropriate in a condition.  Try replacing the '?.' with a '.', testing the left-hand side for null if necessary.
```

And so we do:
```dart
if (foo != null && foo.isAThing)
```

It would be convenient if `foo?.isBar` evaluated to false when `foo` is `null`.

**Comment by jonahwilliams** [comment]
Another note is that if `foo` is a class field, the user needs to write out either:

```dart
if (foo != null && foo!.isBar)
```

or
```dart
var localFoo = foo;
if (localFoo != null && localFoo.isBar)
```

**Comment by DenizAksimsek** [comment]
You could also use 
```dart
if (foo?.isBar ?? false) {
  print("I'm a bar");
}
```

**Comment by zhesha** [comment]
my way is 
```
bool? foo;
if (foo == true) {
  print("I'm a bar");
}
```

**Comment by munificent** [comment]
The [preferred idiom](https://dart.dev/guides/language/effective-dart/usage#do-use--to-convert-null-to-a-boolean-value) for this is `?? false` like @DenizAksimsek suggests.

We spent some time discussing having `null` in a boolean context implicitly behave like `false`. I did some investigation through a large corpus of Dart code. If I remember right, in about 20% of the cases where a nullable boolean expression appeared inside an `if` condition, the default value the user wanted was `true`, not `false`. (In other words, for every four `?? false`, I found one `?? true`.)

To us, that sent a signal that both choices were common enough that having the language silently guess you meant `false` would mask bugs. Instead, there is no default conversion and we make the user choose explicitly. With null safety, this becomes a compile time error, so the user is asked to make that choice quite clearly. It's a little more verbose, but we think the clarity is worth it.
